superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -34,17 +37,103 @@ export const FormFieldOrder = [
   'username',
   'password',
   'database_name',
+  'credentials_info',
 ];
 
 interface FieldPropTypes {
   required: boolean;
   changeMethods: { onParametersChange: (value: any) => string } & {
     onChange: (value: any) => string;
-  };
+  } & { onParametersUploadFileChange: (value: any) => string };
   validationErrors: JsonObject | null;
   getValidation: () => void;
 }
 
+const credentialsInfo = ({
+  required,
+  changeMethods,
+  getValidation,
+  validationErrors,
+}: FieldPropTypes) => {
+  const [uploadOption, setUploadOption] = useState<string>('upload');
+  const [fileToUpload, setFileToUpload] = useState<string>(null);
+  return (
+    <CredentialInfoForm>
+      <>
+        <label className="label-select">
+          How do you want to enter service account credentials?
+        </label>
+        <Select
+          defaultValue={'file'}
+          style={{ width: '100%' }}
+          onChange={option => setUploadOption(option)}
```

Comment:
I believe the arrow function here isn't necessary. You should just be able to pass the `setUploadOption` function since you're passing the same args.